pkg/destroy/azure/azure.go

```diff
@@ -196,6 +205,73 @@ func (o *ClusterUninstaller) Run() error {
 	return utilerrors.NewAggregate(errs)
 }
 
+func deleteAzureStackPublicRecords(ctx context.Context, o *ClusterUninstaller) error {
+	ctx, cancel := context.WithTimeout(ctx, 10*time.Minute)
+	defer cancel()
+
+	logger := o.Logger
+	rgName := o.BaseDomainResourceGroupName
+
+	dnsClient := azurestackdns.NewZonesClientWithBaseURI(o.Environment.ResourceManagerEndpoint, o.SubscriptionID)
+	dnsClient.Authorizer = o.Authorizer
+
+	recordsClient := azurestackdns.NewRecordSetsClientWithBaseURI(o.Environment.ResourceManagerEndpoint, o.SubscriptionID)
+	recordsClient.Authorizer = o.Authorizer
+	clusterName := o.ClusterName
+
+	// collect records from private zones in rgName
+	var errs []error
+
+	zonesPage, err := dnsClient.ListByResourceGroup(ctx, rgName, to.Int32Ptr(100))
+	logger.Debug(err)
+	if err != nil {
+		if zonesPage.Response().IsHTTPStatus(http.StatusNotFound) {
+			logger.Debug("already deleted the AzureStack zones")
+			return utilerrors.NewAggregate(errs)
+		}
+		errs = append(errs, errors.Wrap(err, "failed to list dns zone"))
+		if isAuthError(err) {
+			return err
+		}
+	}
+
+	allZones := sets.NewString()
+	for ; zonesPage.NotDone(); err = zonesPage.NextWithContext(ctx) {
+		if err != nil {
+			errs = append(errs, errors.Wrap(err, "failed to advance to next dns zone"))
+			continue
+		}
+		for _, zone := range zonesPage.Values() {
+			allZones.Insert(to.String(zone.Name))
+		}
+	}
+
+	privateDNSRecordName := fmt.Sprintf("api-int.%s", clusterName)
+	publicDNSRecordName := fmt.Sprintf("api.%s", clusterName)
+	for _, zone := range allZones.List() {
+		for recordPages, err := recordsClient.ListByDNSZone(ctx, rgName, zone, to.Int32Ptr(100), ""); recordPages.NotDone(); err = recordPages.NextWithContext(ctx) {
+			if err != nil {
+				return err
+			}
+			for _, record := range recordPages.Values() {
+				if t := to.String(record.Name); t == privateDNSRecordName || t == publicDNSRecordName {
```

Comment:
I wonder if in addition we can check based on a tag/metadata. I see that there is support for creating with a tag/metadata. 

So would the client support a check to see if the record has the tag with key/value:
```
"kubernetes.io_cluster.${var.cluster_id}" = "owned"
```
I don't believe we can locate the DNS record by this tag, which would be the ideal scenario (then we would not need to add new values to the metadata). 

This can also be followed up in a subsequent PR to add the tag in the terraform code & check the tag here.

Reply:
Actually if we can check on tags, then we wouldn't need/want to check the names. That is, we could just loop through all the records and see if they match the tag (not the name). But yeah that can be a subsequent PR once we confirm installer, ingress and any other dns records can be tagged.

Reply:
I'll check it out. Will update this PR.

Reply:
Are we adding the tags right now? I don't see them in the DNS records on the portal.

Reply:
No. We can add them in a subsequent pr.